src/fmtnum.rs

```diff
@@ -123,3 +123,13 @@ macro_rules! fmtfloat {
             }
         })*)
 }
+
+macro_rules! fmttype {
```

Comment:
is this the right place for this?

Reply:
it generates the definition of what formatter should be used for which type of numbers, sounds good to me, but I'm not afraid to move it elsewhere if you want me to.